File: Navigation/MenuDemo/app/src/main/java/edu/cs4730/menudemo/ActionMenuActivity.java
package edu.cs4730.menudemo;

import android.content.Intent;
import android.os.Bundle;

import androidx.core.app.NavUtils;
import androidx.appcompat.app.AppCompatActivity;

import android.view.Menu;
import android.view.MenuItem;
import android.widget.TextView;

import edu.cs4730.menudemo.databinding.ActionmenuBinding;

/**
 * This activity show how to setup both buttons, and menus (and sub menus).
 * but all the work is in the menu/action.xml file.
 */

public class ActionMenuActivity extends AppCompatActivity {
    ActionmenuBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActionmenuBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        //turn on up button
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.action, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            // Navigate "up" the demo structure to the launchpad activity.
            // See http://developer.android.com/design/patterns/navigation.html for more.
            NavUtils.navigateUpTo(this, new Intent(this, MainActivity.class));
            return true;
        }
        //Normally you would deal with all the menu items here, which I'm ignoring for the purpose of the example.
        return super.onOptionsItemSelected(item);
    }

}

File: Navigation/MenuDemo_kt/app/src/main/java/edu/cs4730/menudemo_kt/ActionMenuActivity.kt
package edu.cs4730.menudemo_kt

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.NavUtils
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import edu.cs4730.menudemo_kt.databinding.ActionmenuBinding

/**
 * This activity show how to setup both buttons, and menus (and sub menus).
 * but all the work is in the menu/action.xml file.
 */
class ActionMenuActivity : AppCompatActivity() {
    lateinit var binding: ActionmenuBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActionmenuBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.mainmenu) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            WindowInsetsCompat.CONSUMED
        }
        //turn on up button
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.action, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            // Navigate "up" the demo structure to the launchpad activity.
            // See http://developer.android.com/design/patterns/navigation.html for more.
            NavUtils.navigateUpTo(this, Intent(this, MainActivity::class.java))
            return true
        }
        //Normally you would deal with all the menu items here, which I'm ignoring for the purpose of the example.
        return super.onOptionsItemSelected(item)
    }
}